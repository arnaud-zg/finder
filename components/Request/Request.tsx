import { inlineBlock, padding } from 'csstips'
import { rem } from 'csx'
import { format } from 'date-fns'
import React from 'react'
import { style } from 'typestyle'
import { colors } from '../../constants/colors'
import { Paragraph } from '../Paragraph'
import { Title } from '../Title'

interface IProps {
  description: string
  identifier: string
  locationIndication: string
  publishTime: string
}

const tagClassname = style(inlineBlock, padding(rem(0.25)), {
  backgroundColor: colors.gray.toHexString(),
  borderRadius: rem(0.25),
  color: colors.light.toHexString(),
  marginLeft: rem(0.25),
  $nest: {
    '&:hover': {
      backgroundColor: colors.lightGray.toHexString(),
      color: colors.dark.toHexString(),
    },
  },
})

const locationIndicationClassName = style(inlineBlock, {
  textDecoration: 'underline',
})

export class Request extends React.Component<IProps> {
  render() {
    const {
      description,
      identifier,
      locationIndication,
      publishTime,
    } = this.props
    return (
      <div id={identifier}>
        <Title content={description} level={3} />
        {!!locationIndication && (
          <div className={locationIndicationClassName}>
            <Paragraph content={locationIndication} />
          </div>
        )}
        <div className={tagClassname}>
          {format(new Date(publishTime), "dd/MM/yyyy 'at' HH:mm")}
        </div>
      </div>
    )
  }
}
