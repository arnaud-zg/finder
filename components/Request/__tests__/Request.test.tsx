import React from 'react'
import renderer from 'react-test-renderer'
import { Request } from '../Request'

describe('Request', () => {
  it('renders correctly', () => {
    const testRenderer = renderer.create(
      <Request
        description="Native English Speaker Seeking Accomodation"
        identifier="3"
        locationIndication="Montmartre - Paris"
        publishTime="2019-09-29T13:50:42.926Z"
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
