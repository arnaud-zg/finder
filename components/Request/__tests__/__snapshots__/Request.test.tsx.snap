// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request renders correctly 1`] = `
<div
  id="3"
>
  <h3
    className="fhwo3dt"
  >
    Native English Speaker Seeking Accomodation
  </h3>
  <div
    className="f4lw1zx"
  >
    <p
      className="f2qtt8h"
    >
      Montmartre - Paris
    </p>
  </div>
  <div
    className="f12o3wr"
  >
    29/09/2019 at 15:50
  </div>
</div>
`;
